Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4131,3 +4131,81 @@ WebKitFeatureList* webkit_settings_get_development_features(void)
 {
     return webkitFeatureListCreate(WebPreferences::internalDebugFeatures());
 }
+
+/**
+ * webkit_settings_apply_from_key_file:
+ * @settings: a #WebKitSettings
+ * @key_file: a #GKeyFile
+ * @error: return location for error or %NULL to ignore
+ *
+ * Reads the contents of the `websettings` group from the given @key_file and apply the value of
+ * each key/value to the corresponding property on the @settings.
+ *
+ * Returns: %TRUE if the settings were correctly applied or %FALSE otherwise.
+ *
+ * Since: 2.44
+ */
+gboolean webkit_settings_apply_from_key_file(WebKitSettings* settings, GKeyFile* keyFile, GError ** error)
```

Comment:
Could we not hardcode the name of the section to be `websettings`, and have a function parameter instead? If we want to have a default section name, we can allow the parameter to be nullable.

```suggestion
gboolean webkit_settings_apply_from_key_file(WebKitSettings* settings, GKeyFile* keyFile, const char* sectionName, GError ** error)
```